addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlActionHandler.java

```diff
@@ -291,8 +297,9 @@ private void executeRollershutterTask() {
         }
         this.waitForEvent = false;
 
-        if (this.rollershutterTask != null) {
-            this.rollershutterTask.run();
+        Runnable runnable = this.rollershutterTask;
+        if (runnable != null) {
+            runnable.run();
```

Comment:
If you call the run method you are actually not starting a thread, if you intend to do so consider submitting it to a / the `scheduler` instead. If you intend to do this I would prefer not (ab)use a `Runnable` here

Reply:
@martinvw The runnable variable got introduced to get rid of the IDE potential null warning. But it did expose a major design flaw that has been there for a while. I am indeed abusing the Runnable, not sure anymore if I intended to have it run in a thread. I will change that part of the code.

Reply:
So let's not change the logic but change how it looks :-)

Lets introduce a small (local) functional interface, to make it clear we dit not intend to create a thread based Runnable:

```java
@FunctionalInterface
interface Action {
  void execute();
}
```

As long as we don't have a parameter and a return there is nothing the JDK offers except for `Runnable` but I see some people that share my opinion that is likely to confuse others and might be altered by someone later intending to do good :-)

See also:

https://stackoverflow.com/questions/23958814/functional-interface-that-takes-nothing-and-returns-nothing
https://stackoverflow.com/questions/27973294/function-with-no-args-and-no-return-type-void-in-java-8?lq=1#comment44339647_27973294
https://stackoverflow.com/questions/23868733/java-8-functional-interface-with-no-arguments-and-no-return-value

Technically the `Runnable` interface is exactly what you need but it looks confusing :-)